File: Benchmarks/BinaryTree/Java/BinaryTree.java
public class BinaryTree {
    public static Node build(int nodes) {
        if (nodes == 0) {
            return null;
        }

        int nodesLeft = nodes / 2;
        int nodesRight = nodes - nodesLeft - 1;

        Node left = build(nodesLeft);
        Node right = build(nodesRight);

        return new Node(left, right);
    }

    public static void count(Node tree, int depth, int[] target) {
        if (tree == null) {
            return;
        }

        target[depth] += 1;
        count(tree.left, depth + 1, target);
        count(tree.right, depth + 1, target);
    }

    public static void main(String[] args) {
        int nodes = Integer.parseInt(args[0]);
        int depth = (int) Math.ceil((double) Math.log(nodes) / (double) Math.log(2.0));

        Node tree = build(nodes);
        int[] counts = new int[depth];

        count(tree, 0, counts);

        for (int i = 0; i < depth; i++) {
            System.out.format("%d %d%n", i, counts[i]);
        }
    }
}

File: Benchmarks/BinaryTree/C#/BinaryTree.cs
public class Node
{
    public Node left;
    public Node right;

    public Node(Node left, Node right)
    {
        this.left = left;
        this.right = right;
    }
}

public static class BinaryTree
{
    public static Node Build(int nodes)
    {
        if (nodes == 0)
        {
            return null;
        }

        int nodesLeft = nodes / 2;
        int nodesRight = nodes - nodesLeft - 1;

        Node left = Build(nodesLeft);
        Node right = Build(nodesRight);

        return new Node(left, right);
    }

    public static void Count(Node node, int depth, int[] counts)
    {
        if (node == null)
            return;

        counts[depth] += 1;
        Count(node.left, depth + 1, counts);
        Count(node.right, depth + 1, counts);
    }

    public static void Main(string[] args)
    {
        int nodes = int.Parse(args[0]);
        int depth = (int)System.Math.Ceiling(System.Math.Log((double)nodes) / System.Math.Log(2.0));

        Node node = Build(nodes);
        int[] counts = new int[depth];

        Count(node, 0, counts);

        for (int d = 0; d < depth; d++)
        {
            System.Console.WriteLine($"{d} {counts[d]}");
        }
    }
}